### PR TITLE
Change `KeyPair` type from `int` to `Hash`

### LINKED ISSUE #1153: Change `KeyPair.from_private_key` argument type from int to Hash

### PR DESCRIPTION
**Pull request by tomek0123456789**
Closes #1153

## Introduced changes

- `KeyPair` constructor accepts `Hash` type
- `KeyPair.from_private_key` accepts `Hash` type

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/net/signer/stark_curve_signer.py [hunk 1]
```diff
@@ -28,8 +29,21 @@ class KeyPair:
...
+        if isinstance(public_key, str):
+            self.public_key = int(public_key, 0)
+        else:
+            self.public_key = public_key
+
     @staticmethod
-    def from_private_key(key: int) -> "KeyPair":
+    def from_private_key(key: Hash) -> "KeyPair":
+        if isinstance(key, str):
+            key = int(key, 0)
```
Comment (drknzz) [inline starknet_py/net/signer/stark_curve_signer.py:46]: this works for "123" as well 👍

File: starknet_py/net/signer/test_stark_curve_signer.py [hunk 2]
```diff
@@ -57,3 +57,29 @@ def test_sign_transaction(transaction):
...
+    assert isinstance(key_pair.public_key, int)
+    assert isinstance(key_pair.private_key, int)
+
+    key_pair = KeyPair.from_private_key("0x123")
+    assert isinstance(key_pair.public_key, int)
+    assert isinstance(key_pair.private_key, int)
+
+    key_pair = KeyPair.from_private_key(0x123)
+    assert isinstance(key_pair.public_key, int)
+    assert isinstance(key_pair.private_key, int)
```
Comment (drknzz) [inline starknet_py/net/signer/test_stark_curve_signer.py:85]: I don't think it's that necessary of a test 😄

File: starknet_py/tests/e2e/docs/quickstart/test_creating_account.py [hunk 3]
```diff
@@ -24,6 +24,8 @@ async def test_creating_account():
 
     # There is another way of creating key_pair
     key_pair = KeyPair.from_private_key(key=123)
+    # or
+    key_pair = KeyPair.from_private_key(key="123")
```
Comment (drknzz) [inline starknet_py/tests/e2e/docs/quickstart/test_creating_account.py:28]: I'd rather show "0x123" if anything 👍

**Review by drknzz — APPROVED** [review]

File: starknet_py/net/signer/test_stark_curve_signer.py (hunk 2 above)
Comment (tomek0123456789) [reply starknet_py/net/signer/test_stark_curve_signer.py:85]: It was supposed to be more of a codecov targeted test, but he didn't like it
image
Will delete it then

**Pull request merged**